lib/bolt/cli.rb

```diff
@@ -477,48 +483,34 @@ def outputter
       @outputter ||= Bolt::Outputter.for_format(@config[:format])
     end
 
-    def load_task_data(name, modulepath)
-      module_name, file_name = name.split('::', 2)
-      file_name ||= 'init'
-
-      begin
-        env = Puppet::Node::Environment.create('bolt', modulepath)
-        Puppet.override(environments: Puppet::Environments::Static.new(env)) do
-          data = Puppet::InfoService::TaskInformationService.task_data(
-            env.name, module_name, name
-          )
-
-          file = data[:files].find { |f| File.basename(f, '.*') == file_name }
-          if file.nil?
-            raise Bolt::CLIError, "Failed to load task file for '#{name}'"
+    def run_task(name, nodes, args, &block)
+      parse_error = nil
+      Puppet.initialize_settings
+      Puppet::Pal.in_tmp_environment('bolt', modulepath: [BOLTLIB_PATH] + @config[:modulepath], facts: {}) do |pal|
+        pal.with_script_compiler do |compiler|
+          begin
+            return compiler.call_function('run_task', name, nodes, args, &block)
+          rescue Puppet::ParseError => e
+            # we assume that the Puppet::ParseError is due to a problem with
+            # the task and/or its parameters, but since the with_script_compiler
+            # method rescues these errors and raises different ones instead,
+            # we save it in a local variable and raise it as Bolt::CLIError
+            # outside of the block
+            parse_error = e
           end
-
-          metadata =
-            if data[:metadata_file]
-              JSON.parse(File.read(data[:metadata_file]))
-            else
-              {}
-            end
-
-          [file, metadata]
         end
-      rescue Puppet::Module::Task::TaskNotFound
-        raise  Bolt::CLIError,
-               "Could not find task '#{name}' in module '#{module_name}'"
-      rescue Puppet::Module::MissingModule
-        # Generate message so we don't expose "bolt environment"
-        raise  Bolt::CLIError, "Could not find module '#{module_name}'"
       end
+      raise Bolt::CLIError, parse_error.message
     end
 
-    def run_plan(plan, args, modulepath)
+    def run_plan(plan, args)
       Dir.mktmpdir('bolt') do |dir|
         cli = []
         Puppet::Settings::REQUIRED_APP_SETTINGS.each do |setting|
           cli << "--#{setting}" << dir
         end
```

Comment:
Should we perform the same kind of setup (i.e. creating a temporary directory, and setting the REQUIRED_APP_SETTINGS) in the run_task method too? Why is it even necessary?